src/simple-length.js

```diff
@@ -26,6 +26,8 @@
           this.value = nValue;
         }
       }
+    } else {
+      throw new TypeError('\'' + type + '\' is not a valid type for a SimpleLength.');
     }
     if (this.value == undefined) {
       throw new TypeError('Value of SimpleLength must be a number or a numeric string.');
```

Comment:
To do that I'd need to re-check everything in the first if statement (i.e. that value is a number or a numeric string). I think it makes more sense here.

I've changed the error check in number-value.js to match this one.